src/Agent.Worker/Build/TfsVCCommandManager.cs

```diff
@@ -236,6 +236,19 @@ private string FormatArguments(FormatFlags formatFlags, params string[] args)
             return string.Join(" ", formattedArgs);
         }
 
+        private string GetEndpointData(ServiceEndpoint endpoint, string name)
```

Comment:
I saw that as well, but I'm not sure of a clean approach around it. The key is that tracing is built-in to these functions. The only other clean-ish option I see - given the current organizational patterns in the agent - is to create a helper:
```csharp
public static class EndpointUtil
{
    public static string GetData(Tracing trace, ServiceEndpoint endpoint, string name) { [...] }
}
```